File: Backend/VestTour.Domain/Entities/OrderDetail.cs
﻿using System;
using System.Collections.Generic;

namespace VestTour.Domain.Entities;

public partial class OrderDetail
{
    public int OrderId { get; set; }

    public int ProductId { get; set; }
   // public string? ProductCode { get; set; }

    public int? Quantity { get; set; }

    public decimal? Price { get; set; }

    public virtual Order Order { get; set; } = null!;

    public virtual Product Product { get; set; } = null!;
}

File: Mobile/mobileapp/lib/models/orderDetail.dart
class OrderDetails {
  int? orderId;
  int? productId;
  int? quantity;
  double? price;

  OrderDetails({this.orderId, this.productId, this.quantity, this.price});

  OrderDetails.fromJson(Map<String, dynamic> json) {
    orderId = json['orderId'];
    productId = json['productId'];
    quantity = json['quantity'];
    price = json['price'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['orderId'] = this.orderId;
    data['productId'] = this.productId;
    data['quantity'] = this.quantity;
    data['price'] = this.price;
    return data;
  }
}